packages/flutter/lib/src/services/text_input.dart

```diff
@@ -805,17 +812,34 @@ RawFloatingCursorPoint _toTextPoint(FloatingCursorDragState state, Map<String, d
   return RawFloatingCursorPoint(offset: offset, state: state);
 }
 
-class _TextInputClientHandler {
-  _TextInputClientHandler() {
-    SystemChannels.textInput.setMethodCallHandler(_handleTextInputInvocation);
+/// A handler for the [SystemChannels.textInput] channel.
+///
+/// This class is used internally by the Flutter SDK and is only visible for
+/// testing.
+@visibleForTesting
+class TextInputClientHandler {
+  /// Creates a [TextInputClientHandler].
+  @visibleForTesting
+  TextInputClientHandler(this._channel, this._currentConnection) : assert(_channel != null) {
```

Comment:
The arguments are only there for testing, right? I wonder if we should default them to what we need them in production.

Reply:
I'm refactoring the test.